qa/rpc-tests/prioritisetransaction.py

```diff
@@ -9,124 +9,222 @@
+            newdict[key]["total_fee"] = dict[key]["fee"]
+            newdict[key]["size"] = dict[key]["size"]
+            if ("priority_delta" in dict[key]):
+                newdict[key]["total_priority"] += dict[key]["priority_delta"]
+            if ("fee_delta" in dict[key]):
+                newdict[key]["total_fee"] += dict[key]["fee_delta"]
+        return sorted(newdict.items(), key=lambda x: x[1]["total_priority" if priority_otherwise_fee else "total_fee"], reverse=True)
+    
+    def get_max_priority_and_max_fee(self, rawmempool):
+        sorted_rawmempool = self.get_sorted_by(rawmempool, True)
+        max_priority = sorted_rawmempool[0][1]["total_priority"]
+        sorted_rawmempool = self.get_sorted_by(rawmempool, False)
+        max_fee = sorted_rawmempool[0][1]["total_fee"]
+        return max_priority, max_fee
+
+    def get_last_tx_selectable_by_priority(self, rawmempool):
+        sorted_rawmempool = self.get_sorted_by(rawmempool, True)
+        block_incremental_size = 1000 # check CreateNewBlock for this value
+        last_tx_selectable_by_priority = None
+        for index in range(len(sorted_rawmempool)):
+            tx_size = sorted_rawmempool[index][1]["size"]
+            if (block_incremental_size + tx_size >= self.blockprioritysize or
+                sorted_rawmempool[index][1]["total_priority"] <= 100000000 * 144 / 250): # check AllowFreeThreshold for this value
+                last_tx_selectable_by_priority = sorted_rawmempool[index][0]
+                break
+            else:
+                block_incremental_size += tx_size
+        return last_tx_selectable_by_priority
+
+    def check_if_included_by_priority(self, tx, last_tx_by_priority, block_template):
+        tx_included_by_priority = False
+        tx_included = False
+        last_tx_by_priority_included = False
+        for index in range(len(block_template["transactions"])):
+            if (block_template["transactions"][index]["hash"] == last_tx_by_priority):
+                last_tx_by_priority_included = True
+            if (block_template["transactions"][index]["hash"] == tx):
+                tx_included = True
+                if (not last_tx_by_priority_included):
+                    tx_included_by_priority = True
+                break
+        return tx_included_by_priority, tx_included
+
+    def check_if_mined_by_priority(self, tx, last_tx_by_priority, block):
+        tx_mined_by_priority = False
+        tx_mined = False
+        last_tx_by_priority_mined = False
+        for index in range(len(block["tx"])):
+            if (block["tx"][index] == last_tx_by_priority):
+                last_tx_by_priority_mined = True
+            if (block["tx"][index] == tx):
+                tx_mined = True
+                if (not last_tx_by_priority_mined):
+                    tx_mined_by_priority = True
+                break
+        return tx_mined_by_priority, tx_mined
+
     def run_test (self):
```

Comment:
This might be a good chance to add a brief high level description as we do for other tests, to give some quick context on the purpose of this test

Reply:
I agree we can take chance of this refactoring/fixing to increase the test documentation; added a bunch of lines at the start of `run_test` method.